biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/bamstats/BamStats.scala

```diff
@@ -182,41 +162,46 @@ object BamStats extends ToolCommand {
   }
 
   /**
-    * This will start the subjobs for each contig and collect [[Stats]] on contig level
-    *
-    * @param region Region to check, mostly yhis is the complete contig
-    * @param bamFile Input bam file
-    * @param binSize stats binsize
-    * @param threadBinSize Thread binsize
-    * @return Output stats
-    */
-  def processContig(region: BedRecord,
-                    bamFile: File,
-                    binSize: Int,
-                    threadBinSize: Int,
-                    outputDir: File): Future[Stats] = Future {
-    val scattersFutures = region
-      .scatter(binSize)
-      .grouped(
-        (region.length.toDouble / binSize).ceil.toInt / (region.length.toDouble / threadBinSize).ceil.toInt)
-      .map(scatters => processThread(scatters, bamFile))
-      .toList
-    waitOnFutures(scattersFutures, Some(region.chr))
-  }
-
-  /**
     * This method will wait when all futures are complete and collect a single [[Stats]] instance
     *
     * @param futures List of futures to monitor
     * @param msg Optional message for logging
     * @return Output stats
     */
-  def waitOnFutures(futures: List[Future[Stats]], msg: Option[String] = None): Stats = {
+  def waitOnFutures(futures: List[Future[Map[BedRecord, Stats]]],
+                    msg: Option[String] = None): (Stats, Map[String, Stats]) = {
     msg.foreach(m => logger.info(s"Start monitoring jobs for '$m', ${futures.size} jobs"))
     futures.foreach(_.onFailure { case t => throw new RuntimeException(t) })
-    val stats = Await.result(Future.sequence(futures).map(_.fold(Stats())(_ += _)), Duration.Inf)
+    val totalSize = futures.size
+    val totalStats = Stats()
+    val contigStats: mutable.Map[String, Stats] = mutable.Map()
+
+    def wait(todo: List[Future[Map[BedRecord, Stats]]]): Unit = {
+      try {
+        logger.info(s"${totalSize - todo.size}/$totalSize tasks done")
+        val completed = todo.groupBy(_.isCompleted)
+        val bla = completed.getOrElse(true, Nil).foreach { f =>
```

Comment:
Maybe something as foundContigStats instead of bla?

Reply:
Did remove it completely, bla was not used anyway. it currently is only a foreach loop